alpha/declcfg/load_test.go

```diff
@@ -92,6 +92,73 @@ func TestLoadReader(t *testing.T) {
 	}
 }
 
+func TestWalkMetasFS(t *testing.T) {
+	type spec struct {
+		name              string
+		fsys              fs.FS
+		assertion         require.ErrorAssertionFunc
+		expectNumPackages int
+		expectNumChannels int
+		expectNumBundles  int
+		expectNumOthers   int
+	}
+	specs := []spec{
+		{
+			name:      "Error/NilFS",
+			fsys:      nil,
+			assertion: require.Error,
+		},
+		{
+			name:      "Error/NonExistentDir",
+			fsys:      os.DirFS("non/existent/dir/"),
+			assertion: require.Error,
+		},
+		{
+			name:      "Error/Invalid",
+			fsys:      invalidFS,
+			assertion: require.Error,
+		},
+		{
+			name:              "Success/ValidDir",
+			fsys:              validFS,
+			assertion:         require.NoError,
+			expectNumPackages: 3,
```

Comment:
Would that leave a testing gap where both implementations are victims of the same underlying bug, so they end up seeing the same data, but its the wrong data?

Reply:
Yeah, six of one, half dozen of the other, I guess - this approach has the bug where the expected packages are incorrect.

Reply:
Fair. To be honest, I cribbed from an existing test. I think [this test](https://github.com/operator-framework/operator-registry/blob/45f491b932997c20ac9c6e54b4b4fa822994cefb/alpha/declcfg/load_test.go#L119) gives us what both of us are asking for. We have lots of overlapping tests because all these functions now basically build on each other.

It wouldn't be an unreasonable argument to mimic that test instead though. Because in this case, we want to see if our `Meta` objects look like we expect. I'll take a look.